src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,603 @@
+          compile_cc.target,
+          [(conf, compile_cc.jar_file) for conf in self._confs])
+      elif target.has_sources('.scala'):
+        self.context.products.get_data('rsc_outline_classpath').add_for_target(
+          rsc_outline_cc.target,
+          [(conf, rsc_outline_cc.rsc_mjar_file) for conf in self._confs])
+      else:
+        pass
+
+  def create_empty_extra_products(self):
+    super(RscCompile, self).create_empty_extra_products()
+
+    compile_classpath = self.context.products.get_data('compile_classpath')
+    classpath_product = self.context.products.get_data('rsc_outline_classpath')
+    if not classpath_product:
+      self.context.products.get_data('rsc_outline_classpath', compile_classpath.copy)
+    else:
+      classpath_product.update(compile_classpath)
+
+  def execute(self):
+    if JvmPlatform.global_instance().get_options().compiler == 'rsc':
+      # Calling BaseZincCompile directly because ZincCompile won't run a compile if
+      # rsc is specified.
+      return BaseZincCompile.execute(self)
+
+  def rsc_outline_key_for_target(self, compile_target):
+    if compile_target.has_sources('.java'):
+      # rsc_outline dependencies on java depends on compile.
+      return self.compile_against_rsc_outline_key_for_target(compile_target)
+    elif compile_target.has_sources('.scala'):
+      return "rsc_outline({})".format(compile_target.address.spec)
+    else:
+      raise TaskError('unexpected target for compiling with rsc_outline .... {}'.format(compile_target))
+
+  def compile_against_rsc_outline_key_for_target(self, compile_target):
+    return "compile_against_rsc_outline({})".format(compile_target.address.spec)
+
+  def create_compile_jobs(self,
+                          compile_target,
+                          compile_contexts,
+                          invalid_dependencies,
+                          ivts,
+                          counter,
+                          runtime_classpath_product):
+    def work_for_vts(vts, ctx, key, classpath_product_key):
+      progress_message = ctx.target.address.spec
+
+      # Double check the cache before beginning compilation
+      hit_cache = self.check_cache(vts, counter)
+
+      if not hit_cache:
+        # Compute the compile classpath for this target.
+        cp_entries = self._zinc.compile_classpath_entries(
+          classpath_product_key,
+          ctx.target,
+          extra_cp_entries=self._extra_compile_time_classpath,
+        )
+        upstream_analysis = dict(self._upstream_analysis(compile_contexts, cp_entries))
+
+        is_incremental = self.should_compile_incrementally(vts, ctx)
+        if not is_incremental:
+          # Purge existing analysis file in non-incremental mode.
+          safe_delete(ctx.analysis_file)
+          # Work around https://github.com/pantsbuild/pants/issues/3670
+          safe_rmtree(ctx.classes_dir)
+
+        dep_context = DependencyContext.global_instance()
+        tgt, = vts.targets
```

Comment:
Is the comma intentional? there's another instance below
```
>>> t, = 5
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not iterable
>>> t, = [5,6]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack
```

Reply:
It comes from upstream. I'm not sure why it's there but I didn't want to change it here without updating there. I think I'll leave it in when I dedupe the above so as not to further confuse the diff.